core/kover/learning/common/models.py

```diff
@@ -21,11 +22,80 @@
 
 conjunction = "conjunction"
 disjunction = "disjunction"
+scm = "scm"
+cart = "cart"
 
 class BaseModel(object):
     def __init__(self):
-        self.rules = []
+        self.rules = None
         super(BaseModel, self).__init__()
+        
+    def predict(self, X):
+        raise NotImplementedError()
+        
+    @property
+    def learner(self):
+        raise NotImplementedError()
+        
+    def __str__(self):
+        return self._to_string()
+        
+class CART_Model(BaseModel):
+    def __init__(self, class_tags=None):
+        super(CART_Model, self).__init__()
+        self.decision_tree = None
+        self.class_tags = class_tags
+        
+    def add_decision_tree(self, tree):
+        self.decision_tree = tree
+        
+    def predict(self, X):
+        if self.decision_tree is None:
+            raise RuntimeError("A decision tree must be fitted prior to calling predict.")
+        predictions = self.decision_tree.predict(X)
+        return np.asarray(predictions, dtype=np.uint8)
+        
+    def predict_proba(self, X):
+        if self.decision_tree is None:
+            raise RuntimeError("A decision tree must be fitted prior to calling predict.")
+        self.decision_tree.predict_proba(X)
+        
+    @property
+    def learner(self):
```

Comment:
That's strange. I don't know why you would need this, but if it's required I don't mind keeping it. Keep in mind that you can always use isinstance(object, class). This also works for parent classes. So say that I have a disjunction model, I would have `isinstance(model, SCMModel) = True` and `isinstance(model, DisjunctionModel) = True`.